pkg/kubectl/cmd/cmd.go

```diff
@@ -125,25 +125,25 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return describer, nil
 		},
-		Printer: func(cmd *cobra.Command, mapping *meta.RESTMapping, noHeaders bool) (kubectl.ResourcePrinter, error) {
+		Printer: func(mapping *meta.RESTMapping, noHeaders bool) (kubectl.ResourcePrinter, error) {
 			return kubectl.NewHumanReadablePrinter(noHeaders), nil
 		},
-		Resizer: func(cmd *cobra.Command, mapping *meta.RESTMapping) (kubectl.Resizer, error) {
+		Resizer: func(mapping *meta.RESTMapping) (kubectl.Resizer, error) {
 			client, err := clients.ClientForVersion(mapping.APIVersion)
 			if err != nil {
 				return nil, err
 			}
 			return kubectl.ResizerFor(mapping.Kind, client)
 		},
-		Reaper: func(cmd *cobra.Command, mapping *meta.RESTMapping) (kubectl.Reaper, error) {
+		Reaper: func(mapping *meta.RESTMapping) (kubectl.Reaper, error) {
 			client, err := clients.ClientForVersion(mapping.APIVersion)
 			if err != nil {
 				return nil, err
 			}
 			return kubectl.ReaperFor(mapping.Kind, client)
 		},
-		Validator: func(cmd *cobra.Command) (validation.Schema, error) {
-			if cmdutil.GetFlagBool(cmd, "validate") {
+		Validator: func() (validation.Schema, error) {
+			if flags.Lookup("validate").Value.String() == "true" {
```

Comment:
I don't know that every command needs to validate.  Get, describe, and a few others definitely do not.


> On Mar 15, 2015, at 7:09 PM, Jeff Lowdermilk <notifications@github.com> wrote:
> 
> In pkg/kubectl/cmd/cmd.go:
> 
> >  			client, err := clients.ClientForVersion(mapping.APIVersion)
> >  			if err != nil {
> >  				return nil, err
> >  			}
> >  			return kubectl.ReaperFor(mapping.Kind, client)
> >  		},
> > -		Validator: func(cmd *cobra.Command) (validation.Schema, error) {
> > -			if cmdutil.GetFlagBool(cmd, "validate") {
> > +		Validator: func() (validation.Schema, error) {
> > +			if flags.Lookup("validate").Value.String() == "true" {
> This is causing a nil dereference in hack/test-cmd.sh. I think the issue is that the validate flag is added to the flagset of each command but not to factory.flags. Can probably fix by adding in BindFlags.
> 
> —
> Reply to this email directly or view it on GitHub.
>